src/Microsoft.OpenApi/Writers/OpenApiWriterBase.cs

```diff
@@ -446,6 +446,10 @@ public void WriteJsonSchema(JsonSchema schema, OpenApiSpecVersion version)
                     {
                         FindJsonSchemaRefs.ResolveJsonSchema(schema);
                     }
+                    else if (Settings.InlineLocalReferences)
```

Comment:
I'm still not convinced that we should be doing anything about inlining in the writer.  All inlining should be done by a "referenceResolver" that mutates the current document.